ip-utils.c

```diff
@@ -0,0 +1,159 @@
+/*! \file    ip-utils.c
+ * \author   Johan Ouwerkerk <jm.ouwerkerk@gmail.com>
+ * \copyright GNU General Public License v3
+ * \brief    IP address related utility functions (headers)
+ * \details  Provides functions to query for network devices with a given device name or address.
+ * Devices may be looked up by either a device name or by the IPv4 or IPv6 address of the configured network interface.
+ * This functionality may be used to bind to user configurable network devices instead of relying on unpredictable implementation defined defaults.
+ * Parsing IPv4/IPv6 addresses is done using inet_pton(), making these functions generally robust against malformed input.
+ *
+ * \ingroup core
+ * \ref core
+ */
+
+#include <errno.h>
+#include <sys/types.h>
+#include <arpa/inet.h>
+#include <net/if.h>
+#include <string.h>
+
+#include "ip-utils.h"
+
+static int compare_byte_arrays(const uint8_t * b1, const uint8_t * b2, const size_t size) {
```

Comment:
Please prefix all methods, even those that are static and only used internally, with something `janus_`. In this case, I guess `janus_iputils_` or `janus_ip_` would be the way to go? Feel free to think of something cooler :smile: